docs/madr/decisions/033-meshhttproute-meshgateway.md

```diff
@@ -0,0 +1,318 @@
+# `MeshHTTPRoute`/`MeshTCPRoute` for `MeshGateway`
+
+- Status: accepted
+
+Technical Story: #8142
+
+## Context and Problem Statement
+
+We have some very similar resources, `MeshHTTPRoute`/`MeshTCPRoute` and `MeshGatewayRoute` that
+both configure route matching, filtering and redirecting semantics.
+This MADR proposes adapting `MeshHTTPRoute` and `MeshTCPRoute` for use with
+`MeshGateway.`
+
+## Decision Drivers <!-- optional -->
+
+- Fewer CRDs
+- Parity with `MeshGatewayRoute`
+
+## Considered Options
+
+- Keep two resources
+- Use `MeshHTTPRoute`/`MeshTCPRoute`
+
+## Decision Outcome
+
+Chosen option:
+
+- Adapt `MeshHTTPRoute`/`MeshTCPRoute`
+- Add `spec.to[].hostnames` to `MeshHTTPRoute`
+- Require `spec.to[].targetRef.kind: Mesh`
+- Require `backendRefs` to be set
+- Don't support policy matching on `MeshHTTPRoute`/`MeshTCPRoute` if it attaches to a gateway
+
+### Hostname
+
+With `MeshGatewayRoute` and `HTTPRoute`, users can set hostnames to only apply
+the route rules to requests going to the given hosts.
+We don't have this possibility with `MeshHTTPRoute`, so we can either:
+
+- add `hostnames` to the `MeshHTTPRoute`, likely directly under `spec.to`
+  entries
+- have users on each rule do hostname matching using the header match on `host`
+
+This MADR proposes adding `spec.to[].hostnames`. Hostname matching is in some sense the
+"highest level" matching possible with HTTP and it's likely that all rules of a
+given resource are intended to apply to a given hostname set, whether that's a
+specific hostname, a wildcard pattern or to all hostnames.
+
+The second option might be feasible but it only makes sense if the cost of having
+host matching separate is very high. It also differs from Gateway API, which is
+a disadvantage.
+
+### `to.targetRef`
+
+When using `MeshHTTPRoute`/`MeshTCPRoute` for service to service communication, it's very
+useful to limit the effects based on request source and destination. We do
+this by setting `spec.targetRef`, the source, and `spec.to[].targetRef`, the destination.
+
+With `MeshGateway`, once a request hits the gateway, the routing resources
+_define_ the destination. We have to put the rules somewhere, so this MADR
+proposes `spec.to`. There's no real semantic value to a distinct
+`spec.to[].targetRef`, or for that matter, a `spec.from[].targetRef`.
+
+This MADR proposes requiring `spec.to[].targetRef.kind: Mesh`.
```

Comment:
Maybe I'm missing something, but it says that it requires `spec.to[].targetRef.kind: Mesh`, but I see no such thing in the examples.

Reply:
Woops, forgot to update the examples